Code/Framework/AzToolsFramework/Tests/Input/QtEventToAzInputMapperTests.cpp

```diff
@@ -0,0 +1,535 @@
+    {
+        // setup
+        const AzFramework::InputChannelId mouseWheelId = AzFramework::InputDeviceMouse::Movement::Z;
+        const char* mouseWheelChannelName = mouseWheelId.GetName();
+
+        AzFramework::InputChannelNotificationBus::Handler::BusConnect();
+        m_captureAzEvents = false;
+
+        SimulateMouseWheelEvent(m_rootWidget.get());
+
+        // qt validation
+        ASSERT_EQ(m_signalEvents.size(), 3);
+
+        EXPECT_EQ(m_signalEvents[0].m_eventType, QEvent::Type::Wheel);
+        EXPECT_EQ(m_signalEvents[0].m_scrollPhase, Qt::ScrollBegin);
+
+        EXPECT_EQ(m_signalEvents[1].m_eventType, QEvent::Type::Wheel);
+        EXPECT_EQ(m_signalEvents[1].m_scrollPhase, Qt::ScrollUpdate);
+
+        EXPECT_EQ(m_signalEvents[2].m_eventType, QEvent::Type::Wheel);
+        EXPECT_EQ(m_signalEvents[2].m_scrollPhase, Qt::ScrollEnd);
+
+        // az validation
+        ASSERT_EQ(m_azChannelEvents.size(), 3);
+
+        EXPECT_STREQ(m_azChannelEvents[0].m_inputChannelId.GetName(), mouseWheelChannelName);
+
+        EXPECT_STREQ(m_azChannelEvents[1].m_inputChannelId.GetName(), mouseWheelChannelName);
+
+        EXPECT_STREQ(m_azChannelEvents[2].m_inputChannelId.GetName(), mouseWheelChannelName);
+
+        // cleanup
+        AzFramework::InputChannelNotificationBus::Handler::BusDisconnect();
+    }
+
+    TEST_F(QtEventToAzInputMapperFixture, MouseWheel_AzHandlerCaptured_ReceivedZeroSignalAndThreeAzChannelEvents)
+    {
+        // setup
+        const AzFramework::InputChannelId mouseWheelId = AzFramework::InputDeviceMouse::Movement::Z;
+        const char* mouseWheelChannelName = mouseWheelId.GetName();
+
+        AzFramework::InputChannelNotificationBus::Handler::BusConnect();
+        m_captureAzEvents = true;
+
+        SimulateMouseWheelEvent(m_rootWidget.get());
+
+        // qt validation
+        EXPECT_EQ(m_signalEvents.size(), 0);
+
+        // az validation
+        ASSERT_EQ(m_azChannelEvents.size(), 3);
+
+        EXPECT_STREQ(m_azChannelEvents[0].m_inputChannelId.GetName(), mouseWheelChannelName);
+
+        EXPECT_STREQ(m_azChannelEvents[1].m_inputChannelId.GetName(), mouseWheelChannelName);
+
+        EXPECT_STREQ(m_azChannelEvents[2].m_inputChannelId.GetName(), mouseWheelChannelName);
+
+        // cleanup
+        AzFramework::InputChannelNotificationBus::Handler::BusDisconnect();
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+    struct MouseButtonIdsParam
+    {
+        Qt::MouseButton m_qt;
+        AzFramework::InputChannelId m_az;
+    };
+
+    class MouseButtonParamQtEventToAzInputMapperFixture
+        : public QtEventToAzInputMapperFixture
+        , public ::testing::WithParamInterface<MouseButtonIdsParam>
+    {
+    };
+
+    TEST_P(MouseButtonParamQtEventToAzInputMapperFixture, MouseClick_NoAzHandlers_ReceivedTwoSignalAndZeroAzChannelEvents)
```

Comment:
suggestion: It might be worth adding some brief comments/descriptions to the tests for added context. The test names are helpful but it's not always clear why what they're testing is important. Some additional context for the reader could be really helpful (especially when one of these inevitably fails in future 😛)